packages/babel-plugin-proposal-decorators/test/fixtures/2023-11-misc/.decorator-evaluation-yield-super-private-all-private/exec.js

```diff
@@ -0,0 +1,33 @@
+{
+  let B;
+
+  class CaptureFactory {
+    static *id(fn) {
+      yield fn;
+    }
+  }
+
+  function dummy() {}
+
+  class A extends CaptureFactory {
+    static #X = "A#X";
+    static *[Symbol.iterator] () {
+      B = @dummy
+      class B {
+        static #X = "B#X";
+        @(yield* super.id(_ => _.#X), dummy) #method () {}
+        @(yield* super.id(_ => _.#X), dummy) static #Method () {}
+        @(yield* super.id(_ => _.#X), dummy) get #getter () {}
+        @(yield* super.id(_ => _.#X), dummy) static get #Getter () {}
+        @(yield* super.id(_ => _.#X), dummy) set #setter (v) {}
+        @(yield* super.id(_ => _.#X), dummy) static set #Setter (v) {}
+        @(yield* super.id(_ => _.#X), dummy) #property;
+        @(yield* super.id(_ => _.#X), dummy) static #Property;
+        @(yield* super.id(_ => _.#X), dummy) accessor #accessor;
+        @(yield* super.id(_ => _.#X), dummy) static accessor #Accessor;
+      }
+    }
+  }
+
+  expect([...A].map(fn => fn(B)).join(",")).toBe("B#X,B#X,B#X,B#X,B#X,B#X,B#X,B#X,B#X,B#X");
+}
```

Comment:
This edge case test is disabled because I can't figure out how to transform it.

Reply:
In this case, can we introduce a new public element and delete it immediately after defined?
```js
class C {
  // for demo purpose, the `delete` expression can be directly inserted into staticInitializerAssignments
  @(function(_, context) { context.addInitializer(function() { delete this[temp] } })
  static [/* decorator evaluations */, temp = Symbol()];
}
```
Of course using `delete` is a performance killer, but this approach will be only invoked when there is no public class elements, which should be quite rare.

Reply:
That sounds good to me.
We can even do it when someone needs it. :)

Reply:
> for delete, let's put it in a static { delete this._ } block at the beginning of the class?

I plan to insert the delete expression right before the `applyDecs` call, since its parent block is already the first class element.

Reply:
This test is now enabled and merged with [decorator-evaluation-yield-private-super-property/exec.js](https://github.com/babel/babel/pull/16325/commits/707b91ce50f88b8f185d613c456bc1c4d600b287#diff-7dbf2473f4fc123ed19f0ce92ba8f3765e93a7a9fe856052b15e64f05a67f2e0)

It also happens to catch a bug in the `helper-replace-supers`, currently Babel is incorrectly replacing the `super` in the decorators of the given class method. This should have been handled in the `environmentVisitor` but the root class method path is not visited because the `visitSelf` argument of `traverse.node` is not exposed to the `path.traverse`.

https://github.com/babel/babel/pull/16325/commits/707b91ce50f88b8f185d613c456bc1c4d600b287